### PR TITLE
Remove unused codes from Spring Cloud Azure

### PR DESCRIPTION
**Pull request by saragluna**
This PR will remove unused codes from the `KeyVaultOperation` and `KeyVaultPropertySource`.

### CODE REVIEW

File: sdk/spring/spring-cloud-azure-autoconfigure/src/main/java/com/azure/spring/cloud/autoconfigure/keyvault/environment/KeyVaultOperation.java [hunk 1]
```diff
@@ -205,21 +203,4 @@ void setProperties(HashMap<String, String> properties) {
         this.properties = properties;
     }
 
-    boolean isUp() {
```
Comment (chenrujun) [inline sdk/spring/spring-cloud-azure-autoconfigure/src/main/java/com/azure/spring/cloud/autoconfigure/keyvault/environment/KeyVaultOperation.java:208]: In main branch, it's used for KeyVaultHealthIndicator. Why KeyVaultHealthIndicator is deleted in 4.0 branch?

Comment (saragluna) [reply sdk/spring/spring-cloud-azure-autoconfigure/src/main/java/com/azure/spring/cloud/autoconfigure/keyvault/environment/KeyVaultOperation.java:208]: we use another method for kv health indicator now, and we do not have the kv health indicator for property source now.

Comment (chenrujun) [reply sdk/spring/spring-cloud-azure-autoconfigure/src/main/java/com/azure/spring/cloud/autoconfigure/keyvault/environment/KeyVaultOperation.java:208]: I see.
Now for key vault secrets, we only have `KeyVaultSecretHealthIndicator` which only check health of `SecretAsyncClient`.

We can add health indicator for property source if customer ask for this feature in the future.

**Review by chenrujun — APPROVED** [review]
LGTM.

File: sdk/spring/spring-cloud-azure-autoconfigure/src/main/java/com/azure/spring/cloud/autoconfigure/keyvault/environment/KeyVaultOperation.java (hunk 1 above)
Comment (chenrujun) [reply sdk/spring/spring-cloud-azure-autoconfigure/src/main/java/com/azure/spring/cloud/autoconfigure/keyvault/environment/KeyVaultOperation.java:208]: Could you please add this into changelog?

Comment (saragluna) [reply sdk/spring/spring-cloud-azure-autoconfigure/src/main/java/com/azure/spring/cloud/autoconfigure/keyvault/environment/KeyVaultOperation.java:208]: will do

**Pull request merged**